src/vs/workbench/browser/parts/views/views.ts

```diff
@@ -698,6 +699,27 @@ export class ViewsService extends Disposable implements IViewsService {
 
 		return null;
 	}
+
+	getProgressIndicator(id: string): IProgressIndicator | undefined {
+		const viewContainer = this.viewDescriptorService.getViewContainer(id);
+		if (viewContainer === null) {
+			return undefined;
+		}
+
+		const location = this.viewContainersRegistry.getViewContainerLocation(viewContainer);
+
+		let viewPaneContainer;
+		if (location === ViewContainerLocation.Sidebar) {
+			const viewlet = this.viewletService.getInstantiatedViewlet(viewContainer.id);
+			viewPaneContainer = viewlet?.getViewPaneContainer();
+		} else if (location === ViewContainerLocation.Panel) {
+			const panel = this.panelService.getInstantiatedPanel(viewContainer.id);
+			viewPaneContainer = (panel as IPaneComposite)?.getViewPaneContainer();
+		}
```

Comment:
Not happy with exposing `getInstantiatedViewlet` or `getInstantiatedPanel` , feel like exposing some internals. I understand the necessity but I think this can be done with out this.

Why can't you scope `getProgressIndicator` to active view container? Why do you need progress indicator for a view that is not visible?

Also `ViewPaneContainers` are registered in Views service for the service listen to view pane containers events. If wanted `ViewsService` can also maintain the registry of `ViewPaneContainers` per `ViewContainer. We would do this anyway in future when we remove panels and viewlets.

https://github.com/microsoft/vscode/blob/55c2dc54e296c2b5e2d297b7f3df2329a19b5794/src/vs/workbench/browser/parts/views/views.ts#L489-L495

Reply:
Why is it any more internal than the active viewlet?

A view can start progress in the background when its not currently visible/expanded, and then become visible and ideally the progress would show up.

I'm not sure I'm following what you are saying about `ViewPaneContainers`

Reply:
I do not like `activeViewlet` too. But it does not mean that we can add more internal methods.

> A view can start progress in the background when its not currently visible/expanded, and then become visible and ideally the progress would show up.

Is not it not performat to do some activity when view is not shown.. view is taking some cpu cycles which is not needed. 

> I'm not sure I'm following what you are saying about ViewPaneContainers

Please check the code I mentioned above. Every `ViewPaneContainer` is registered in ViewsService. So you can get `ViewPaneContainer` for a given `ViewContainer` without going to Viewlet or Panel service. Lets try not to add more stuff to it as we want to move away from them.

Reply:
@sandy081 views (such as scm) still indicate progress in the background today because they are updating based on git actions. Timeline falls in the same boat and I am not sure we should make any such limitation. Even if a view is not able to render background progress, it would still be able to do it. Thus, allowing progress to be shown can help reveal a misbehaving extension.

Reply:
I see. As suggested please reuse and build on top of `registerViewPaneContainer` instead of exposing internals.

https://github.com/microsoft/vscode/blob/55c2dc54e296c2b5e2d297b7f3df2329a19b5794/src/vs/workbench/browser/parts/views/views.ts#L489-L495

Reply:
OK, pushed changes to avoid exposing anything via the `getInstantiated*` methods